server/mscalendar/oauth2.go

```diff
@@ -72,6 +95,18 @@ func (app *oauth2App) CompleteOAuth2(authedUserID, code, state string) error {
 		return err
 	}
 
+	uid, err := app.Store.LoadMattermostUserId(me.ID)
+	if err == nil {
+		user, userErr := app.PluginAPI.GetMattermostUser(uid)
+		if userErr == nil {
+			app.Poster.DM(authedUserID, RemoteUserAlreadyConnected, me.Mail, user.Username)
+			return errors.Errorf(RemoteUserAlreadyConnected, me.Mail, user.Username)
+		} else {
+			// Orphaned connected account. Let it be overwritten by passing through here?
+			app.Poster.DM(authedUserID, RemoteUserAlreadyConnectedNotFound, me.Mail)
```

Comment:
@levb This is a weird situation, where the remote account has already been connected to a MM account, but the MM account failed to be retrieved from the MM server. Should we reject this connection attempt in this case? Allowing the new connection leaves the KV store in a weird state where the `mattermostUserIDKV` is overwritten, but the orphaned user still exists in `userKV` and `userIndexKV`.

I'm thinking we should reject it because the failure to retrieve the user could have been for any reason.